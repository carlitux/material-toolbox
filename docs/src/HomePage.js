// @flow
/* eslint-disable */
import * as React from 'react';
import Highlighter from './Highlighter';
import ThemedLink from './ThemedLink';

import { LayoutCell } from 'material-toolbox/layout-grid';
import { Text } from 'material-toolbox/typography';
/* eslint-enable */

const importCode = '@import "@material/drawer/mdc-drawer";';

const HomePage = () => (
  <LayoutCell align="bottom" span={12}>
    <Text component="div" textStyle="body2">
      <p>Second note.- Most of examples, design and content is from:</p>
      <p>
        <ThemedLink
          to="https://material.io/components/web/catalog/"
          target="_blank">
          https://material.io/components/web/catalog/
        </ThemedLink>
      </p>
      <p>
        <ThemedLink to="https://react-mdc.github.io/" target="_blank">
          https://react-mdc.github.io/
        </ThemedLink>
      </p>
    </Text>
    <Text component="h2" textStyle="headline">
      Getting started
    </Text>
    <Text component="p" textStyle="body1">
      Get up and running with React Material Components web
    </Text>
    <Highlighter language="bash">
      $ npm install --save material-toolbox
    </Highlighter>
    <Text component="h3" textStyle="title">
      NOTE.- You have to import the css/scss files from material components,
      into your scss or the css as needed.
    </Text>
    <Highlighter language="css">{importCode}</Highlighter>
  </LayoutCell>
);

export default HomePage;
