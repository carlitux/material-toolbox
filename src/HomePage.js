// @flow
import * as React from 'react';

import LayoutCell from './material-toolbox/layout-grid/Cell';
import Text from './material-toolbox/typography/Text';

import Highlighter from './Highlighter';
import ThemedLink from './ThemedLink';

const importCode = '@import "@material/drawer/mdc-drawer";';

const HomePage = () => (
  <LayoutCell align="bottom" span={12}>
    <Text component="div" textStyle="body2">
      <p>Second note.- Most of examples, design and content is from:</p>
      <p>
        <ThemedLink
          to="https://material.io/components/web/catalog/"
          target="_blank">
          https://material.io/components/web/catalog/
        </ThemedLink>
      </p>
      <p>
        <ThemedLink to="https://react-mdc.github.io/" target="_blank">
          https://react-mdc.github.io/
        </ThemedLink>
      </p>
    </Text>

    <Text component="h2" textStyle="headline">
      Getting started
    </Text>

    <Text component="p" textStyle="body1">
      Get up and running with React Material Components web
    </Text>

    <Highlighter language="bash">
      $ npm install --save material-toolbox
    </Highlighter>

    <Text component="h2" textStyle="headline">
      Notes
    </Text>

    <p>
      1. Some components needs to refresh on first load of app, this because
      those components needs to know the correct values of DOM and those
      components have a property <strong>refreshAt</strong> to get the timeout
      to be refreshed. This value depends a lot of your app. If you need to
      avoid that, use another library like{' '}
      <ThemedLink
        target="_blank"
        to="https://github.com/ctrlplusb/react-sizeme">
        react-sizeme
      </ThemedLink>
    </p>

    <p>
      2. You have to import the css/scss files from material components, into
      your scss or the css as needed.
    </p>

    <Highlighter language="css">{importCode}</Highlighter>
  </LayoutCell>
);

export default HomePage;
