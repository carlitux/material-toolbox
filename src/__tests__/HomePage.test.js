import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../HomePage';

describe('component::HomePage', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<HomePage />);
    expect(component).toMatchSnapshot();
  });
});
