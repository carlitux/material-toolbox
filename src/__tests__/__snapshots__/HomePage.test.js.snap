// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::HomePage Should match snapshoot 1`] = `
<Cell
  align="bottom"
  span={12}
>
  <Text
    adjustMargin={false}
    component="div"
    textStyle="body2"
  >
    <p>
      Second note.- Most of examples, design and content is from:
    </p>
    <p>
      <ThemedLink
        component="a"
        target="_blank"
        to="https://material.io/components/web/catalog/"
      >
        https://material.io/components/web/catalog/
      </ThemedLink>
    </p>
    <p>
      <ThemedLink
        component="a"
        target="_blank"
        to="https://react-mdc.github.io/"
      >
        https://react-mdc.github.io/
      </ThemedLink>
    </p>
  </Text>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="headline"
  >
    Getting started
  </Text>
  <Text
    adjustMargin={false}
    component="p"
    textStyle="body1"
  >
    Get up and running with React Material Components web
  </Text>
  <Highlighter
    language="bash"
  >
    $ npm install --save material-toolbox
  </Highlighter>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="headline"
  >
    Notes
  </Text>
  <p>
    1. Some components needs to refresh on first load of app, this because those components needs to know the correct values of DOM and those components have a property 
    <strong>
      refreshAt
    </strong>
     to get the timeout to be refreshed. This value depends a lot of your app. If you need to avoid that, use another library like
     
    <ThemedLink
      component="a"
      target="_blank"
      to="https://github.com/ctrlplusb/react-sizeme"
    >
      react-sizeme
    </ThemedLink>
  </p>
  <p>
    2. You have to import the css/scss files from material components, into your scss or the css as needed.
  </p>
  <Highlighter
    language="css"
  >
    @import "@material/drawer/mdc-drawer";
  </Highlighter>
</Cell>
`;
